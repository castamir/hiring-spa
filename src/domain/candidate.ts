enum LinkType {
  LINKED_IN,
  SMITIO,
  MEET_FRANK,
  TECHLOOP,
  STARTUP_JOBS,
}

enum State {
  NEW_CONTACT,
  APPROVED,
  DECLINED,
  INITIAL_MEETING,
  TECHNICAL_INTERVIEW,
  TEST_ASSIGNMENT_SEND,
  TEST_ASSIGNMENT_IN_REVIEW,
  FINAL_INTERVIEW,
  HIRED,
  SHORTLIST,
}

type CandidateBase = {
  name: string;
  surname: string;
  email: string;
  links: { type: LinkType; link: string }[];
  state: State;
  location: string;
  nationality: string;
};

export type CandidateReadDto = CandidateBase & {
  id: string;
};
export type CandidateCreateDto = CandidateBase;
