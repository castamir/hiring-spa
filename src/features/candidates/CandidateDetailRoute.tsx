import React from "react";
import { CandidateReadDto } from "../../domain/candidate";

export type Props = {
  candidate: CandidateReadDto | null;
};

const CandidateDetailRoute: React.FC<Props> = ({ candidate }) => {
  if (!candidate) {
    return null;
  }

  return (
    <div>
      <h2>
        {candidate.name} {candidate.surname}
      </h2>
      <div>
        <p>{candidate.nationality}</p>
        <p>{candidate.state}</p>
        <p>{candidate.email}</p>
        <p>{candidate.links}</p>
        <p>{candidate.location}</p>
      </div>
    </div>
  );
};

export default CandidateDetailRoute;
