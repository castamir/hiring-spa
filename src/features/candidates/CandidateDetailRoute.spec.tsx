import { shallow } from "enzyme";
import React from "react";

import CandidateDetailRoute from "./CandidateDetailRoute";
import { State } from "../../domain/candidate";

describe("<CandidateDetailRoute />", () => {
  it("should render without candidate", () => {
    const wrapper = shallow(<CandidateDetailRoute />);
    expect(wrapper.html()).toEqual(null);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with candidate", () => {
    const candidateExample = {
      id: "ID 1",
      email: "jedna@domain.com",
      name: "John",
      surname: "Doe",
      state: State.NEW_CONTACT,
      nationality: "Czech",
      links: [],
      location: "Brno",
    };
    const wrapper = shallow(
      <CandidateDetailRoute candidate={candidateExample} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h2")).toHaveLength(1);
    expect(wrapper.find("h2").exists()).toBeTruthy(); //toBe(true)
    expect(wrapper.find("h2").length).toEqual(1);

    expect(wrapper.find("h2").first().text()).toContain("John");
    expect(wrapper.find("h2").first().text()).toContain(
      candidateExample.surname
    );
    // expect(wrapper.text()).not.toContain("Brno");
  });
});
