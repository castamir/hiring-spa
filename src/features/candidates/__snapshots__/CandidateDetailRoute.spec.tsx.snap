// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CandidateDetailRoute /> should render with candidate 1`] = `
<div>
  <h2>
    John
     
    Doe
  </h2>
  <div>
    <p>
      Czech
    </p>
    <p>
      0
    </p>
    <p>
      jedna@domain.com
    </p>
    <p />
    <p>
      Brno
    </p>
  </div>
</div>
`;

exports[`<CandidateDetailRoute /> should render without candidate 1`] = `""`;
